### PR TITLE
Creating a storage_management decorator 

### LINKED ISSUE #167: Raise something on storage abort
We need to raise something when storage aborts.

I was just trying to save and save and save from webview with nothing actually saving, but from browser terminal my post and put requests were coming back as 200 Ok.

### PR DESCRIPTION
**Pull request by rich-hart**
handles storage persistence, error catching and logging, as well as 503 service unavailable responses.

Fix #167

### CODE REVIEW

File: cnxauthoring/tests/test_functional.py [hunk 1]
```diff
@@ -2799,6 +2799,51 @@ def test_user_contents_hide_documents_inside_binders(self):
...
+
+        self.addCleanup(setattr, storage, 'conn',
+                        psycopg2.connect(storage.conn.dsn))
+
+        storage.conn.close()
+
+        response = self.testapp.post_json(
+            '/users/contents',
+            {'title': u'My document タイトル'},
+            status=201,
```
Comment (karenc) [inline cnxauthoring/tests/test_functional.py]: Shouldn't this be 503?

**Comment by karenc** [comment]
Looks good to me other than the test status

**Comment by rich-hart** [comment]
Great!  Just changed the test status to 503.  Let me know if you notice anything else I should update!

**Comment by rich-hart** [comment]
Great! Updating status to 503

**Pull request merged**